### ISSUE TITLE
Unable to initialize IAM sub-system

### ISSUE THREAD

**Issue by cniackz** [minio/minio#16839]
## Summary:

Changing Root Password is exposing this message:

```
API: SYSTEM()
Time: 13:17:08 UTC 03/17/2023
DeploymentID: 97e999bb-8cbe-4c10-b733-5615b8586306
Error: Unable to initialize IAM sub-system, some users may not be available: signature is invalid (*fmt.wrapError)
       3: internal/logger/logger.go:258:logger.LogIf()
       2: cmd/iam.go:326:cmd.(*IAMSys).Init()
       1: cmd/server-main.go:653:cmd.serverMain.func3()
```

Public MinIO Slack Conversation: https://minio.slack.com/archives/C3NDUB8UA/p1679053144087879

## Expected Behavior

It would be ideal or expected that no issue is observed in the logs after changing the password. And that all users and policies in the system works well even if root password change.

## Current Behavior

When Changing the Root Password, there is a message in the logs:

```
API: SYSTEM()
Time: 13:17:08 UTC 03/17/2023
DeploymentID: 97e999bb-8cbe-4c10-b733-5615b8586306
Error: Unable to initialize IAM sub-system, some users may not be available: signature is invalid (*fmt.wrapError)
       3: internal/logger/logger.go:258:logger.LogIf()
       2: cmd/iam.go:326:cmd.(*IAMSys).Init()
       1: cmd/server-main.go:653:cmd.serverMain.func3()
```

But more testing is required to determine if this is going to affect other accounts.

## Possible Solution

Not at this moment.

## Steps to Reproduce (for bugs)

https://github.com/cniackz/public/wiki/Changing-root-password-in-MinIO-Baremetal

1. Run latest MinIO in bare metal
2. Add one bucket and one file via Console 
3. Stop MinIO (Ctrl+c)
4. Set new root password but keep the same root user and start MinIO again, you will see the log.

## Context

It is affecting community because they used to have shorter password and for security reasons now they decided to make password longer but the log is not giving confidence that this is working well...

## Regression

Observed in version: `RELEASE.2023-03-13T19-46-17Z`

## Your Environment

* Version used (`minio --version`): `RELEASE.2023-03-13T19-46-17Z`
* Server setup and configuration: 1 server 1 disk for testing only
* Operating System and version (`uname -a`):
```
$ uname -a
Darwin Cesars-MacBook-Pro.local 22.3.0 Darwin Kernel Version 22.3.0: Thu Jan  5 20:48:54 PST 2023; root:xnu-8792.81.2~2/RELEASE_ARM64_T6000 arm64
```

**Comment by vadmeste** [comment]
@cniackz this was fixed with this PR, https://github.com/minio/minio/pull/16812

**Comment by harshavardhana** [comment]
This is fixed in #16812 it is regression - the fix has relevant details.
